src/vs/platform/extensionManagement/common/extensionsProfileScannerService.ts

```diff
@@ -247,14 +248,23 @@ export abstract class AbstractExtensionsProfileScannerService extends Disposable
 						this.reportAndThrowInvalidConentError(file);
 					}
 					let location: URI;
-					if (isString(e.location)) {
+					if (isString(e.relativeLocation)) {
+						// Extension in new format. No migration needed.
+						location = this.resolveExtensionLocation(e.relativeLocation);
+					} else if (isString(e.location)) {
+						// Extension in intermediate format. Migrate to new format.
 						location = this.resolveExtensionLocation(e.location);
+						migrate = true;
+						e.relativeLocation = e.location;
+						// retain old format so that old clients can read it
+						e.location = location.toJSON();
 					} else {
```

Comment:
I'd add a check `else if (isObject(e.location))` check, in case a user made changes, or the file is corrupted

Reply:
Ok, take it back, better add a check in `isStoredProfileExtension`

Reply:
Added check and also added tests